text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,85 @@
+# Sampling Threshold Propagation in TraceState
+
+## Motivation
+
+Sampling can theoretically take place at nearly any point in a distributed tracing system. If sampling is to be performed at multiple points in the process, the only way to reason about it effectively is to make sure that the sampling decisions are **consistent**. In this context consistency is the property that allows different points in the sampling chain to make the same sampling decisions, based on information included with the trace.
+
+## Explanation
+
+The existing, experimental [specification for probability sampling using TraceState](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-probability-sampling.md) is limited to powers-of-two probabilities, and is designed to work without making assumptions about TraceID randomness.
+This system can only achieve non-power-of-two sampling using interpolation between powers of two, which is unnecessarily restrictive.
+In existing sampling systems, sampling probabilities like 1%, 10%, and 75% are common, and it should be possible to express these without interpolation.
+There is also a need for consistent sampling in the collection path (outside of the head-sampling paths) and using inherent randomness in the traceID is a less-expensive solution than referencing a custom `r-value` from the tracestate in every span.
+This proposal allows for the continued expression of randomness using `r-value` as specified there using the key `r`.
+However, that value is limited to powers of two, while this proposal is not; to distinguish the cases, this proposal uses the key `rv`.
+
+
+In order to make consistent sampling decisions across the entire path of the trace, two values SHOULD be propagated with the trace:
+
+1. A random (or pseudo-random) value of at least 56 bits, called `R` below.
+2. A 56-bit trace threshold as expressed in the TraceState, called `T` below.
+
+The sampling decision is propagated with the following algorithm:
+* If the `th` key is not specified, then no previous sampling decision has been made.
+* If the value of the `th` key is `!`, always sample.
+* Else parse the `th` key as a hex value as described below.
+* Compare the 56 bits of `T` with the 56 bits of `R`. If `T <= R`, then do not sample.
+* This implies that if `T` is 0, then never sample.
+
+The `R` value MUST be derived as follows:
+* If the Random Trace ID Flag is `true` in the traceparent header, then `R` is the lowest-order 56 bits of the trace-id.
+* Else if the key `rv` is present in the Tracestate header, then `R = rv`.
+* Else if the key `r` is present in the Tracestate header, then `R = 2**(56-r)`.
+* Else `R` should be generated as a random value in the range `(0, (2**56)-1)` and added to the Tracestate header with key `rv`.
```

Comment:
I am wondering if it would be better to require the generation of `R` in this case. Then consistent sampling would always work if the Random Trace ID Flag is not set.